### PR TITLE
approval based

### PR DESCRIPTION
**Pull request by cdonnay**
-approval and bloc plurality

### CODE REVIEW

File: tests/elections/election_types/approval/test_approval.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        profile = PreferenceProfile(ballots=[Ballot(scores={"A": 3})])
+        Approval(profile, m=1)
+
+    with pytest.raises(TypeError):  # must be non-negative
+        profile = PreferenceProfile(ballots=[Ballot(scores={"A": -3})])
+        Approval(profile, m=2)
+
+    with pytest.raises(TypeError):  # must have scores
+        profile = PreferenceProfile(ballots=[Ballot()])
+        Approval(profile, m=2)
```
Comment (peterrrock2) [inline tests/elections/election_types/approval/test_approval.py:148]: Check strings

File: tests/elections/election_types/approval/test_bloc_plurality.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        profile = PreferenceProfile(ballots=[Ballot(scores={"A": 3})])
+        BlocPlurality(profile, m=1)
+
+    with pytest.raises(TypeError):  # must be non-negative
+        profile = PreferenceProfile(ballots=[Ballot(scores={"A": -3})])
+        BlocPlurality(profile, m=1)
+
+    with pytest.raises(TypeError):  # must have scores
+        profile = PreferenceProfile(ballots=[Ballot()])
+        BlocPlurality(profile, m=2)
```
Comment (peterrrock2) [inline tests/elections/election_types/approval/test_bloc_plurality.py:154]: Check strings

**Review by peterrrock2 — CHANGES_REQUESTED** [review]
Small notes in line with the last PR

File: src/votekit/elections/election_types/approval/approval.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+    def __init__(
+        self,
+        profile: PreferenceProfile,
+        m: int = 1,
+        k: Optional[int] = None,
+        tiebreak: Optional[str] = None,
+    ):
+        if not k:
+            k = len(profile.candidates)
```
[the remark this answers is not in the document]
Comment (cdonnay) [reply src/votekit/elections/election_types/approval/approval.py:49]: great catch!

**Review by peterrrock2 — APPROVED** [review]

**Pull request merged**